packages/ui/src/helper.ts

```diff
@@ -454,10 +454,15 @@ const handleTypeChange = (
       delete schema[key as keyof typeof schema];
     }
   });
+  // Apply attributes from new defaultSchema
   const propPanel = Object.values(pluginsRegistry).find(
     (plugin) => plugin?.propPanel.defaultSchema.type === value
   )?.propPanel;
-  Object.assign(schema, propPanel?.defaultSchema || {});
+  Object.keys(propPanel?.defaultSchema || {}).forEach((key) => {
```

Comment:
this was the real bug - we were saving the attributes we wanted above but then overwriting them with the defaults from the new schema (which includes `position { x: 0, y: 0 }`).

I think this worked before because of the order of events. 

It triggered change on `type` first, then (accidentally) triggered change on `x` and `y` so the values were reset to 0 but then quickly put back to what they were before.